core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala

```diff
@@ -291,4 +332,22 @@ private[spark] object HadoopFSUtils extends Logging {
     resolvedLeafStatuses
   }
   // scalastyle:on argcount
+
+  /** A serializable variant of HDFS's BlockLocation. */
```

Comment:
Maybe add in a comment that says we keep these for Hadoop 2.7 combat so we don't make the same mistake next time?

Reply:
Good point - will do after the current CI run finishes.